### PR TITLE
Improve bezier computation performance.

### PR DESCRIPTION
**Pull request by jonahwilliams**
On a Pixel 7 Pro (Skia or Impeller) this change represents about a 2x difference in UI thread work.

For example, running the animated example before:

And After

The reason:

Vector Math's Vector2 class is really slow to allocate and throw away, since its backed by a small Float32List. Generally that class only performs well if you're doing in place mutation. Since we're doing math where we create and throw away a lot of Offsets/vectors, switching to the dart:ui Offset class (which is just a class with two doubles) will perform much better as the compiler can inline the allocations, or even remove them completely.

Finally, removes the usage of the sync* iterator in place of just writing the values directly into the float32 list.

There are still additional opportunities for performance improvements, namely that the Float32Lists can be recycled once a ui.Vertices object is constructed from them.

### CODE REVIEW

File: flutter/mesh/lib/src/internal_stuff/tessellated_mesh.dart [hunk 1]
```diff
@@ -28,96 +23,54 @@ class TessellatedMesh {
...
     required RenderedOMeshRect textureMesh,
     required int tessellation,
-    required bool impellerCompatibilityMode,
   }) {
-    final verticesTrianglesIterable = _getTriangles(
+    final lengthInF32 = _vertexData.length ~/ 2;
+    final offsetInBytes = lengthInF32 * 4;
+    final verticesTriangles = Float32List.view(_vertexData.buffer, 0, lengthInF32);
+    final textureTriangles = Float32List.view(_vertexData.buffer, offsetInBytes, lengthInF32);
+
```
Comment (renancaraujo) [inline flutter/mesh/lib/src/internal_stuff/tessellated_mesh.dart:31]: TIL

File: flutter/mesh/lib/src/internal_stuff/tessellated_mesh.dart [hunk 2]
```diff
@@ -28,96 +23,54 @@ class TessellatedMesh {
     required RenderedOMeshRect verticesMesh,
     required RenderedOMeshRect textureMesh,
     required int tessellation,
-    required bool impellerCompatibilityMode,
```
Comment (renancaraujo) [inline flutter/mesh/lib/src/internal_stuff/tessellated_mesh.dart:31]: We need to keep this on at least until we have a Flutter release with impeller working

**Review by renancaraujo — COMMENTED** [review]
Hey  @jonahwilliams Thanks for your pull request this performance gain looks amazing!

My question is how much do the changes on `OVertex` impact those gains? Also, the bezier control points should be mutable as well, since the non widget contexts, like flame, mutable APIs can be quite handy.

Now, all the changes on `internal_stuff` LGTM 😍

File: flutter/mesh/lib/src/internal_stuff/tessellated_mesh.dart (hunk 2 above)
Comment (jonahwilliams) [reply flutter/mesh/lib/src/internal_stuff/tessellated_mesh.dart:31]: YEah makes sense, but while I was at it I couldn't help myself. :)

**Comment by jonahwilliams** [comment]
Let me pull some numbers. Avoiding the Vector class from vector_math is pretty important.

**Review by renancaraujo — APPROVED** [review]
I am gonna merge it and adjust the public API,
Thanks for this, helps a lot

**Pull request merged**